src/rust/graph-query/src/lib.rs

```diff
@@ -0,0 +1,626 @@
+        _var_allocator: &mut VarAllocator,
+    ) {
+        if self.negated {
+            filter_string.push_str("NOT(");
+        }
+        filter_string.push_str("eq(");
+        filter_string.push_str(property_name);
+        filter_string.push(',');
+        filter_string.push_str(&self.to.to_string());
+        filter_string.push(')');
+        if self.negated {
+            filter_string.push(')');
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct UidEq {
+    to: u64,
+}
+
+impl UidEq {
+    pub fn new(to: u64) -> Self {
+        Self { to }
+    }
+
+    pub fn eq(to: u64) -> Self {
+        Self { to }
+    }
+}
+
+impl PropertyFilter for UidEq {
+    fn to_filter(
+        &self,
+        _property_name: &str,
+        filter_string: &mut String,
+        _var_allocator: &mut VarAllocator,
+    ) {
+        filter_string.push_str("uid(");
+        filter_string.push_str(&self.to.to_string());
+        filter_string.push(')');
+    }
+}
+
+#[derive(Clone)]
+pub struct IntLt {
+    to: u64,
+    negated: bool,
+}
+
+impl IntLt {
+    pub fn new(to: u64, negated: bool) -> Self {
+        Self { to, negated }
+    }
+}
+
+impl PropertyFilter for IntLt {
+    fn to_filter(
+        &self,
+        property_name: &str,
+        filter_string: &mut String,
+        _var_allocator: &mut VarAllocator,
+    ) {
+        if self.negated {
+            filter_string.push_str("NOT(");
+        }
+        filter_string.push_str("lt(");
+        filter_string.push_str(property_name);
+        filter_string.push(',');
+        filter_string.push_str(&self.to.to_string());
+        filter_string.push(')');
+        if self.negated {
+            filter_string.push(')');
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct IntGt {
+    to: u64,
+    negated: bool,
+}
+
+impl IntGt {
+    pub fn new(to: u64, negated: bool) -> Self {
+        Self { to, negated }
+    }
+}
+
+impl PropertyFilter for IntGt {
+    fn to_filter(
+        &self,
+        property_name: &str,
+        filter_string: &mut String,
+        _var_allocator: &mut VarAllocator,
+    ) {
+        if self.negated {
+            filter_string.push_str("NOT(");
+        }
+        filter_string.push_str("gt(");
+        filter_string.push_str(property_name);
+        filter_string.push(',');
+        filter_string.push_str(&self.to.to_string());
+        filter_string.push(')');
+        if self.negated {
+            filter_string.push(')');
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct StrEq {
+    to: String,
+    negated: bool,
+}
+
+impl StrEq {
+    pub fn new(to: impl Into<String>, negated: bool) -> Self {
+        Self {
+            to: to.into(),
+            negated,
+        }
+    }
+
+    pub fn eq(to: impl Into<String>) -> Self {
+        Self {
+            to: to.into(),
+            negated: false,
+        }
+    }
+
+    pub fn neq(to: impl Into<String>) -> Self {
+        Self {
+            to: to.into(),
+            negated: true,
+        }
+    }
```

Comment:
The operation is equality - that's the `eq` in dgraph. It can be negated. I could instead make `neq` `negated` ?

`new` can be removed, it's just a basic constructor.

Reply:
Gonna just leave new fwiw. The other operators are implemented in terms of it, and it could be nicer to write:
`StrEq::new("prop", bool_expr)` vs doing the if/else.

Reply:
> The operation is equality - that's the `eq` in dgraph. It can be negated. I could instead make `neq` `negated` ?

I get that... I'm talking about the fact that you have a struct called "equal" that you create by calling a function called "not equal". It's the same kind of dissonance as saying you want an Angus beef burger, and then being told you can get one at the vegan restaurant down the street.

Reply:
Rather than worry about this, I'm gonna just remove the neq/eq etc helpers for now.